File: T5/IlyinMA/5.cpp
#include<stdio.h>
#include<vector>
using namespace std;

void printVector(vector<int>& a)
{
    printf("Vector:[");
    for(auto it = a.begin(); it != a.end(); ++it)
    {
        if(it != a.begin()) printf(",");
        printf("%d", *it);
    }
    printf("]\n");
}

void radixSort(vector<int>& a)
{
    bool maxLength = false;
    int placement = 1;
    while(!maxLength)
    {
        maxLength = true;
        vector<vector<int>> buckets;
        buckets.resize(10);

        for(size_t i = 0; i < a.size(); i++)
        {
            int tmp = a[i] / placement;
            buckets[tmp % 10].push_back(a[i]);
            if(maxLength && tmp > 0) maxLength = false;
        }

        size_t idx = 0;
        for(size_t b = 0; b < 10; b++)
        {
            for(auto& it : buckets[b])
            {
                a[idx++] = it;
            }
        }
        placement *= 10;
    }
}


int main()
{
    vector<int> a = {35,1,754,65,235,90,967,63,90,323,45};
    printVector(a);
    radixSort(a);
    printVector(a);
    return 0;
}

File: T5/SemenovaDS/5.py
import random
import time

def radixsort( alist ):
    RADIX=10
    maxLength= False
    tmp , placement = -1, 1
    
    while not maxLength:
        maxLength=True
        buckets = [list() for _ in range ( RADIX )]
        for i in alist:
            tmp=i//placement
            buckets[tmp % RADIX].append( i )
            if maxLength and tmp>0:
                maxLength= False
        a=0
        for b in range ( RADIX ):
            buck = buckets[b]
            for i in buck:
                alist[a]= i
                a+=1
        placement*=RADIX
    return alist
arr=[]
for i in range (10001):
    arr.append(random.randint(0, 1000001))
start_time = time.time()
arr=radixsort(arr)
finish_time = time.time()
print ("time :")
print (finish_time-start_time)
